source/MaterialXRender/ShaderRenderer.cpp

```diff
@@ -35,7 +35,12 @@ ShaderRenderer::ShaderRenderer(unsigned int width, unsigned int height, Image::B
     float fW = fH * 1.0f;
     _camera = Camera::create();
     _camera->setViewMatrix(Camera::createViewMatrix(DEFAULT_EYE_POSITION, DEFAULT_TARGET_POSITION, DEFAULT_UP_VECTOR));
-    _camera->setProjectionMatrix(Camera::createPerspectiveMatrix(-fW, fW, -fH, fH, DEFAULT_NEAR_PLANE, DEFAULT_FAR_PLANE));
+
+    #if defined (__APPLE__)
```

Comment:
We might also want to make this Metal specific so that it can still work with OpenGL based MacOS
@jstone-lucasfilm: what would you recommend for the cleanest way for supporting both here?
Perhaps we can move the MaterialXView specific customization to a reusable place:
https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXView/CMakeLists.txt#L8-L27

Reply:
I'm open to that option, @ashwinbhat, though I'm imagining that there may still be value in providing a default camera for simple use cases.  As long as we continue to provide the option of overriding the default camera, it seems reasonable to make this default as suitable as possible for each platform.

Reply:
@nicolassavva-autodesk I like the suggestion of moving the user's OpenGL/Metal selection into a pre-processor definition that is set in CMakeLists.txt and accessible in MaterialXRender.

This could then be used to select between the two perspective matrix options without needing to check platform flags such as `__APPLE__` in cross-platform classes like ShaderRenderer.

What are your thoughts, @Morteeza and @ashwinbhat?

Reply:
@nicolassavva-autodesk What if we were to pass a perspective matrix convention boolean into the constructor for `ShaderRenderer`, allowing the caller to state which convention should be used for the default perspective matrix?  This would allow the platform-independent ShaderRenderer class to remain agnostic of this platform detail, without requiring us to remove the default camera from ShaderRender entirely.

Reply:
I introduced an enum to select between OpenGL and Metal (in lieu of a boolean)
This should suffice for now and we should eventually consider propagating the changes into a future refactor of the `Camera` class and the MaterialXViewer